tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+                "nglcec           = "+self.glc_nec + "\n"
+                "mksrf_fsoitex    = "+self.input_path+"mksrf_soitex.10level.c201018.nc"+"\n"
+                "mksrf_forganic   = "+self.input_path+"mksrf_organic_10level_5x5min_ISRIC-WISE-NCSCD_nlev7_c120830.nc"+"\n"
+                "mksrf_flakwat    = "+self.input_path+"mksrf_LakePnDepth_3x3min_simyr2004_csplk_c151015.nc"+"\n"
+                "mksrf_fwetlnd    = "+self.input_path+"mksrf_lanwat.050425.nc"+"\n"
+                "mksrf_fmax       = "+self.input_path+"mksrf_fmax_3x3min_USGS_c120911.nc"+"\n"
+                "mksrf_fglacier   = "+self.input_path+"mksrf_glacier_3x3min_simyr2000.c120926.nc"+"\n"
+                "mksrf_fvocef     = "+self.input_path+"mksrf_vocef_0.5x0.5_simyr2000.c110531.nc" +"\n"
+                "mksrf_furbtopo   = "+self.input_path+"mksrf_topo.10min.c080912.nc"+"\n"
+                "mksrf_fgdp       = "+self.input_path+"mksrf_gdp_0.5x0.5_AVHRR_simyr2000.c130228.nc"+"\n"
+                "mksrf_fpeat      = "+self.input_path+"mksrf_peatf_0.5x0.5_AVHRR_simyr2000.c130228.nc"+"\n"
+                "mksrf_fsoildepth = "+self.input_path+"mksf_soilthk_5x5min_ORNL-Soil_simyr1900-2015_c170630.nc"+"\n"
+                "mksrf_fabm       = "+self.input_path+"mksrf_abm_0.5x0.5_AVHRR_simyr2000.c130201.nc"+"\n"
+                "outnc_double     = .true. \n"
+                "all_urban        = .false.\n"
+                "no_inlandwet     = .true. \n"
+                "mksrf_furban     = "+self.input_path+"mksrf_urban_0.05x0.05_simyr2000.c170724.nc"+"\n"
+                "gitdescribe      = "+label+"\n"
+                "mksrf_ftopostats = "+self.input_path+"mksrf_topostats_1km-merge-10min_HYDRO1K-merge-nomask_simyr2000.c130402.nc"+"\n"
+                "mksrf_fvegtyp    = "+self.input_path + "pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c170629/mksrf_landuse_histclm50_LUH2_1850.c170629.nc"+"\n"
+                "mksrf_fsoicol    = "+self.input_path + "pftcftlandusedynharv.0.25x0.25.MODIS.simyr1850-2015.c170412/mksrf_soilcolor_CMIP6_simyr2005.c170623.nc"+"\n"
+                "mksrf_flai       = "+self.input_path + "pftcftlandusedynharv.0.25x0.25.MODIS.simyr1850-2015.c170412/mksrf_lai_78pfts_simyr2005.c170413.nc" +"\n"
+                "fdyndat          = ''\n"
+                "numpft           = "+self.num_pft+"\n"
+                "dst_mesh_file    = "+self.input_path+dst_mesh+"\n"
+                "\n&transient\n"
+                "use_transient    = "+use_transient + "\n"
+                "start_year       = "+self.start_year.__str__() + "\n"
+                "end_year         = "+self.end_year.__str__() + "\n"
+                "mksrf_dyn_lu     = "+self.input_path+ "pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c170629" + "\n"
+                "mksrf_fdynuse    = "+self.lu_fname + "\n"
```

Comment:
This section needs some corrections. For example I ran `./mksurfdata.pl -years 1850-2000 -res 4x5` and:
- mksrf_dyn_lu doesn't exist
- mksrf_fdynuse  = 'landuse_timeseries_hist_78pfts_CMIP6_simyr1850-2015.txt'
- fdyndat        = 'landuse.timeseries_4x5_hist_78pfts_CMIP6_simyr1850-2015_c210703.nc'

Reply:
I am not sure what you mean here. Do you mean there is an error with pl code?
When I run `./gen_mksurf_namelist -r 4x5 --sy 1850 --ey 2000`
The name list has:
mksrf_dyn_lu     = /glade/p/cesm/cseg/inputdata/lnd/clm2/rawdata/pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c170629
mksrf_fdynuse    = landuse_timeseries_hist_78pfts_simyr1850-2000.txt